src/helper.js

```diff
@@ -537,41 +539,35 @@ class helper {
 
 		let mosaicsFieldObject = [];
 
-		for (const resolvedMosaic of resolvedMosaics) {
-			if (resolvedMosaic.id.toHex() === http.networkCurrency.mosaicId) {
-				mosaicsFieldObject.push(this.networkCurrencyMosaicBuilder(resolvedMosaic));
-			} else {
-				if (0 < mosaicInfos.length && 0 < mosaicNames.length) {
-					let { divisibility } = mosaicInfos.find(info => info.mosaicId === resolvedMosaic.id.toHex());
-
-					mosaicsFieldObject.push({
-						...resolvedMosaic,
-						rawAmount: resolvedMosaic.amount,
-						mosaicId: resolvedMosaic.id.toHex(),
-						amount: helper.formatMosaicAmountWithDivisibility(resolvedMosaic.amount, divisibility),
-						mosaicAliasName: MosaicService.extractMosaicNamespace({ mosaicId: resolvedMosaic.id.toHex() }, mosaicNames)
-					});
-				}
-			}
-		}
+		const uniqueMosaicIds = [...new Set(resolvedMosaics.map(mosaic => mosaic.id.toHex()))];
 
-		return mosaicsFieldObject;
-	}
+		uniqueMosaicIds.forEach(idHex => {
+			const mosaics = resolvedMosaics.filter(mosaic => mosaic.id.toHex() === idHex);
 
-	static networkCurrencyMosaicBuilder = mosaic => {
-		if (!(mosaic instanceof Mosaic))
-			throw new Error('It required Mosaic instance.');
+			const sumAmount = mosaics.reduce((acc, cur) => acc + Number(cur.amount.toString()), 0);
 
-		if (mosaic.id.toHex() !== http.networkCurrency.mosaicId)
-			throw new Error('Mosaic id does not match network Currency.');
+			if (idHex === http.networkCurrency.mosaicId) {
+				mosaicsFieldObject.push({
+					...mosaics[0],
+					rawAmount: UInt64.fromUint(sumAmount),
+					mosaicId: mosaics[0].id.toHex(),
+					amount: this.formatMosaicAmountWithDivisibility(sumAmount, http.networkCurrency.divisibility),
+					mosaicAliasName: http.networkCurrency.namespaceName
+				});
+			} else {
```

Comment:
why do you need special casing for network currency? why doesn't normal resolution logic work?

Reply:
I excluded this because network currency info doesn't need a query from the network, so we can reduce 1 request for that (better perf).

In the `else` case, we requested mosaic info from the network.

Reply:
that makes sense, but why `sumAmount` in `if` case but `mosaics[0].amount` in `else case?

Reply:
Good catch, it should in both.
I did some refactoring, and an extra unit test, to ensure those are covered.